### DISCUSSION TITLE
How to access privateRuntimeConfig variables from server / api ?

### DISCUSSION THREAD

**Discussion by thibautSchmutz** [nuxt/nuxt#18025] [Questions] [answered]
Hi everyone, 
I declared a private env variable in the nuxt.config.ts : 

```
export default defineNuxtConfig({
  privateRuntimeConfig: {
    apiSecret: "SecretABCD1234" || process.env.API_SECRET,
  },
});
```

And I would like to use it in my server / api files. The documentation says **"The value of this object is accessible from server only using $config"**. Although, I tried to access the $config object using the useRuntimeConfig hook from '#app', but since it is an external package for the server, it doesn't allow me to use it. 

how can I access the $config object from the server ? 

Thank you all !

**Comment by thibautSchmutz** [comment]
MORE INFO : 

I tried this in **server / api / get-config.ts:** 

```
import type { IncomingMessage, ServerResponse } from "http";
import { useRuntimeConfig } from "#app";

export default async (req: IncomingMessage, res: ServerResponse) => {
  const config = useRuntimeConfig();

  console.log(config.apiSecret);

  res.statusCode = 200;
  res.end("Does the env var show in the server console ?");
};
```

Once I make a request to the endpoint, the console returns : *'#app' is imported by server/api/get-config.ts, but could not be resolved – treating it as an external dependency *

**Comment by danielroe** [comment, marked as answer]
At the moment, within server middleware, you can do:

```js
import config from "#config"
```

This may change in future to be more aligned with the Vue composable. See https://github.com/nuxt/framework/pull/625 for context.

**Reply by thibautSchmutz** [reply]
Thank you Daniel ! it works perfectly.

**Comment by ahmedh01** [comment]
Follow-up on this, how do I access the `apiSecret` in my client-side Vue component? Or is this not possible/allowed due to best practices?

Thanks!

**Reply by danielroe** [reply]
https://github.com/nuxt/framework/discussions/1000#discussioncomment-1483083
